src/Azure.Functions.Cli/Actions/LocalActions/CreateProxyAction.cs

```diff
@@ -0,0 +1,209 @@
+                .WithDescription("Template name")
+                .Callback(t => TemplateName = t);
+
+            return Parser.Parse(args);
+        }
+
+        public override Task RunAsync()
+        {
+            if (string.IsNullOrEmpty(ProxyName))
+            {
+                ColoredConsole
+                    .Error
+                    .WriteLine(ErrorColor("Command must specify --name."));
+                return Task.CompletedTask;
+            }
+
+            if (!string.IsNullOrEmpty(Route) && !string.IsNullOrEmpty(TemplateName))
+            {
+                ColoredConsole
+                    .Error
+                    .WriteLine(ErrorColor("Only one of --route or --template values can be specified."));
+                return Task.CompletedTask;
+            }
+
+            IEnumerable<string> templates = null;
+
+            if (string.IsNullOrEmpty(Route) &&
+                string.IsNullOrEmpty(TemplateName))
+            {
+                if (Console.IsOutputRedirected || Console.IsInputRedirected)
+                {
+                    ColoredConsole
+                        .Error
+                    .WriteLine(ErrorColor("Running with stdin\\stdout redirected. Command must specify either --route or --template explicitly."))
+                    .WriteLine(ErrorColor("See 'func help function' for more details"));
+                    return Task.CompletedTask;
+                }
+                else
+                {
+                    templates = _proxyManager.Templates;
+                    ColoredConsole.Write("Select a template: ");
+                    TemplateName = DisplaySelectionWizard(templates);
+                    ColoredConsole.WriteLine(TitleColor(TemplateName));
+                }
+            }
+
+
+            if (!string.IsNullOrEmpty(TemplateName))
+            {
+                if (templates == null)
+                {
+                    templates = _proxyManager.Templates;
+                }
+
+                if (!templates.Contains(TemplateName))
+                {
+                    ColoredConsole.Error.WriteLine(ErrorColor($"Can't find template \"{TemplateName}\" "));
+                    return Task.CompletedTask;
+                }
+
+                _proxyManager.AddProxy(ProxyName, TemplateName);
+                return Task.CompletedTask;
+            }
+
+            ProxyDefinition proxyDefinition = new ProxyDefinition();
+
+            proxyDefinition.Condition = new MatchCondition();
+            proxyDefinition.Condition.Route = Route;
+
+            proxyDefinition.BackendUri = BackendUrl;
+
+            if (Methods != null && Methods.Any())
+            {
+                proxyDefinition.Condition.HttpMethods =  Methods.ToArray() ;
+            }
+
+            _proxyManager.AddProxy(ProxyName, proxyDefinition);
+
+            return Task.CompletedTask;
+        }
+
+        private static T DisplaySelectionWizard<T>(IEnumerable<T> options)
```

Comment:
can you move this to some helper class instead of duplicating it, or did you make changes for your scenario?

Reply:
Oh was planning to do this :-)